building src/pkcs11.c fails with src/pkcs11.c:296:21: error: unused function 'auth_min_rw_so' [-Werror,-Wunused-function]

My CFLAGS are "-Wall -Wextra -pipe -O3 -flto".  Compiling src/pkcs11.lo fails with
```
depbase=`echo src/pkcs11.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ./libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I.. -I./src/lib    -I../src -I../src/lib -Wall -Wextra -Werror -Wformat -Wformat-security -Wstack-protector -fstack-protector-all -Wstrict-overflow=5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fPIC -fPIE  -I/usr/local/include -I/usr/local/include -pthread -I/usr/local/include -Wall -Wextra -pipe -O3 -flto -MT src/pkcs11.lo -MD -MP -MF $depbase.Tpo -c -o src/pkcs11.lo ../src/pkcs11.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./src/lib -I../src -I../src/lib -Wall -Wextra -Werror -Wformat -Wformat-security -Wstack-protector -fstack-protector-all -Wstrict-overflow=5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fPIC -I/usr/local/include -I/usr/local/include -pthread -I/usr/local/include -Wall -Wextra -pipe -O3 -flto -MT src/pkcs11.lo -MD -MP -MF src/.deps/pkcs11.Tpo -c ../src/pkcs11.c  -fPIC -DPIC -o src/.libs/pkcs11.o
../src/pkcs11.c:296:21: error: unused function 'auth_min_rw_so' [-Werror,-Wunused-function]
static inline CK_RV auth_min_rw_so(session_ctx *ctx) {
                    ^
1 error generated.
make: *** [Makefile:1313: src/pkcs11.lo] Error 1
```

--- a/src/pkcs11.c
+++ b/src/pkcs11.c
@@ -292,19 +292,6 @@ static inline CK_RV auth_min_rw_user(session_ctx *ctx) {
     return CKR_USER_NOT_LOGGED_IN;
 }
 
-
-static inline CK_RV auth_min_rw_so(session_ctx *ctx) {
-
-    CK_STATE state = session_ctx_state_get(ctx);
-    switch(state) {
-    case CKS_RW_SO_FUNCTIONS:
-        return CKR_OK;
-        /* no default */
-    }
-
-    return CKR_USER_NOT_LOGGED_IN;
-}
-
 static inline CK_RV auth_any_logged_in(session_ctx *ctx) {
     CK_STATE state = session_ctx_state_get(ctx);
     switch(state) {
@@ -381,11 +368,6 @@ static inline CK_RV auth_init_pin_state(session_ctx *ctx) {
  * Does what __TOKEN_WITH_LOCK_BY_SESSION does, and checks that the session is at least RO User. Ie user logged in and R/W session.
  */
 #define TOKEN_WITH_LOCK_BY_SESSION_USER_RW(userfunc, session, ...) __TOKEN_WITH_LOCK_BY_SESSION(auth_min_rw_user, userfunc, session, ##__VA_ARGS__)
-
-/*
- * Does what __TOKEN_WITH_LOCK_BY_SESSION does, and checks that the session is at least RW So. Ie so logged in and R/W session.
- */
-#define TOKEN_WITH_LOCK_BY_SESSION_SO_RW(userfunc, session, ...) __TOKEN_WITH_LOCK_BY_SESSION(auth_min_rw_so, userfunc, session, ##__VA_ARGS__)
 
 /*
  * Does what __TOKEN_WITH_LOCK_BY_SESSION does, and checks that the session is at least RO User. Ie user or so logged in and R/O or R/W session.

